### PR TITLE
Combine metadata fields

### PR DESCRIPTION
**Pull request by apstndb**
* Combine `scan_type` and `scan_target`
* Combine `call_type`, `iterator_type`, `scan_type` into node title
* Hide `subquery_cluster_node` because `spanner-cli` doesn't display node id

Discussed in https://github.com/cloudspannerecosystem/spanner-cli/issues/57#issuecomment-634399999

### Example 1 Distributed Cross Apply(Back Join)

```sql
EXPLAIN SELECT * FROM Songs@{FORCE_INDEX=SongsBySongName}
```

#### Before
```
Query_Execution_Plan (EXPERIMENTAL)

.
+- Distributed Union (subquery_cluster_node: 1)
    +- Distributed Cross Apply (subquery_cluster_node: 15)
        +- [Input]  Create Batch 
        |   +- Distributed Union (subquery_cluster_node: 4, call_type: Local)
        |       +- Compute Struct 
        |           +- Scan (scan_type: IndexScan, scan_target: SongsBySongName, Full scan: true)
        +- [Map]  Serialize Result 
            +- Cross Apply 
                +- [Input]  Scan (scan_type: BatchScan, scan_target: $v2)
                +- [Map]  Distributed Union (call_type: Local, subquery_cluster_node: 23)
                    +- FilterScan 
                        +- Scan (scan_target: Songs, scan_type: TableScan)
```

#### After

```
Query_Execution_Plan (EXPERIMENTAL)

.
+- Distributed Union 
    +- Distributed Cross Apply 
        +- [Input]  Create Batch 
        |   +- Local Distributed Union 
        |       +- Compute Struct 
        |           +- Index Scan (Full scan: true, Index: SongsBySongName)
        +- [Map]  Serialize Result 
            +- Cross Apply 
                +- [Input]  Batch Scan (Batch: $v2)
                +- [Map]  Local Distributed Union 
                    +- FilterScan 
                        +- Table Scan (Table: Songs)
```

### Example 2 Aggregation and Sort Limit

```sql
EXPLAIN SELECT SingerId, COUNT(*) AS SongCount
FROM Songs
GROUP BY SingerId
ORDER BY SongCount DESC
LIMIT 1
```

#### Before

```
Query_Execution_Plan (EXPERIMENTAL)

.
+- Serialize Result 
    +- Sort Limit (call_type: Global)
        +- Distributed Union (subquery_cluster_node: 3)
            +- Sort Limit (call_type: Local)
                +- Aggregate (iterator_type: Stream)
                    +- Distributed Union (subquery_cluster_node: 6, call_type: Local)
                        +- Scan (scan_type: IndexScan, Full scan: true, scan_target: SongsBySingerAlbumSongNameDesc)
```
#### After

```
Query_Execution_Plan (EXPERIMENTAL)

.
+- Serialize Result 
    +- Global Sort Limit 
        +- Distributed Union 
            +- Local Sort Limit 
                +- Stream Aggregate 
                    +- Local Distributed Union 
                        +- Index Scan (Index: SongsBySingerAlbumSongNameDesc, Full scan: true)
```

### CODE REVIEW

**Comment by apstndb** [comment]
Add basic test cases from the examples.

**Comment by apstndb** [comment]
It seems better to use `google.golang.org/protobuf` because `github.com/golang/protobuf` v1.4.0 provides only alias.

File: query_plan.go [hunk 1]
```diff
@@ -104,15 +105,48 @@ func getMetadataString(node *Node, key string) (string, bool) {
...
+
+func getAllMetadataString(node *Node) string {
+	metadata := node.PlanNode.GetMetadata().GetFields()
 
 	fields := make([]string, 0)
-	for k, v := range node.PlanNode.Metadata.Fields {
-		fields = append(fields, fmt.Sprintf("%s: %s", k, v.GetStringValue()))
+	for k, v := range metadata {
+		switch k {
+		case "call_type", "iterator_type": // Skip because it is displayed in node title
```
Comment (yfuruyama) [inline query_plan.go:131]: It looks like implementations of getNodeTitle() and getAllMetadataString() are coupled with each other.
(getAllMetadataString() depends on the implementation of getNodeTitle()).

How about merging them into a single function?
```
func (n *Node) String() string {
  // assemble operator
  ...
  // assemble metadata
  ...
  return operator + metadata
}
```

Comment (apstndb) [reply query_plan.go]: Fixed in https://github.com/cloudspannerecosystem/spanner-cli/pull/63/commits/8d7a1b4193141c1dbd0590472b2525f9b4dc1582

**Pull request merged**